tests/bgp/test_bgp_update_timer.py

```diff
@@ -0,0 +1,351 @@
+NEIGHBOR_ASN1 = 61001
+NEIGHBOR_PORT0 = 11000
+NEIGHBOR_PORT1 = 11001
+
+
+def _write_variable_from_j2_to_configdb(duthost, template_file, **kwargs):
+    save_dest_path = kwargs.pop("save_dest_path", "/tmp/temp.j2")
+    keep_dest_file = kwargs.pop("keep_dest_file", False)
+    duthost.host.options["variable_manager"].extra_vars.update(kwargs)
+    duthost.template(src=template_file, dest=save_dest_path)
+    duthost.shell("sonic-cfggen -j %s --write-to-db" % save_dest_path)
+    if not keep_dest_file:
+        duthost.file(path=save_dest_path, state="absent")
+
+
+class BGPNeighbor(object):
+
+    def __init__(self, duthost, ptfhost, name, iface,
+                 neighbor_ip, neighbor_asn,
+                 dut_ip, dut_asn, port, is_quagga=False):
+        self.duthost = duthost
+        self.ptfhost = ptfhost
+        self.ptfip = ptfhost.host.options["inventory_manager"].get_host(
+            ptfhost.hostname).vars["ansible_host"]
+        self.iface = iface
+        self.name = name
+        self.ip = neighbor_ip
+        self.asn = neighbor_asn
+        self.peer_ip = dut_ip
+        self.peer_asn = dut_asn
+        self.port = port
+        self.is_quagga = is_quagga
+
+    def start_session(self):
+        """Start the BGP session."""
+        logging.debug("start bgp session %s", self.name)
+        self.ptfhost.shell("ifconfig %s %s/32" % (self.iface, self.ip))
+        self.ptfhost.exabgp(
+            name=self.name,
+            state="started",
+            local_ip=self.ip,
+            router_id=self.ip,
+            peer_ip=self.peer_ip,
+            local_asn=self.asn,
+            peer_asn=self.peer_asn,
+            port=self.port
+        )
+        if not wait_tcp_connection(self.ptfhost, self.ptfip, self.port):
+            raise RuntimeError("Failed to start BGP neighbor %s" % self.name)
+
+        _write_variable_from_j2_to_configdb(
+            self.duthost,
+            "bgp/templates/neighbor_metadata_template.j2",
+            save_dest_path=NEIGHBOR_SAVE_DEST_TMPL % self.name,
+            neighbor_name=self.name,
+            neighbor_lo_addr=self.ip,
+            neighbor_mgmt_addr=self.ip,
+            neighbor_hwsku=None,
+            neighbor_type="ToRRouter"
+        )
+
+        _write_variable_from_j2_to_configdb(
+            self.duthost,
+            "bgp/templates/bgp_template.j2",
+            save_dest_path=BGP_SAVE_DEST_TMPL % self.name,
+            db_table_name="BGP_NEIGHBOR",
+            peer_addr=self.ip,
+            asn=self.asn,
+            local_addr=self.peer_ip,
+            peer_name=self.name
+        )
+
+        if self.is_quagga:
+            allow_ebgp_multihop_cmd = (
+                "vtysh "
+                "-c 'configure terminal' "
+                "-c 'router bgp %s' "
+                "-c 'neighbor %s ebgp-multihop'"
+            )
+            allow_ebgp_multihop_cmd %= (self.peer_asn, self.ip)
+            self.duthost.shell(allow_ebgp_multihop_cmd)
```

Comment:
Need to undo this change on DUT after test?

Reply:
this is for the new neighbors, after test, those new neighbors are removed.